pkg/netflow/flowaggregator/aggregator.go

```diff
@@ -119,6 +124,56 @@ func (agg *FlowAggregator) sendFlows(flows []*common.Flow) {
 	}
 }
 
+func (agg *FlowAggregator) sendExporterMetadata(flows []*common.Flow, flushTime time.Time) {
+	// exporterMap structure: map[NAMESPACE]map[EXPORTER_ID]metadata.NetflowExporter
+	exporterMap := make(map[string]map[string]metadata.NetflowExporter)
+
+	// orderedExporterIDs is used to build predictable metadata payload (consistent batches and orders)
+	// orderedExporterIDs structure: map[NAMESPACE]EXPORTER_ID
+	orderedExporterIDs := make(map[string][]string)
+
+	for _, flow := range flows {
+		exporterIpAddress := common.IPBytesToString(flow.DeviceAddr)
+		if exporterIpAddress == "" || strings.HasPrefix(exporterIpAddress, "?") {
+			log.Errorf("Invalid exporter Addr: %s", exporterIpAddress)
+			continue
+		}
+		exporterID := flow.Namespace + ":" + exporterIpAddress
+		if _, ok := exporterMap[flow.Namespace]; !ok {
+			exporterMap[flow.Namespace] = make(map[string]metadata.NetflowExporter)
+		}
+		if _, ok := exporterMap[flow.Namespace][exporterID]; ok {
+			// this exporter is already in the map, no need to reprocess it
+			continue
+		}
+		exporterMap[flow.Namespace][exporterID] = metadata.NetflowExporter{
+			ID:        exporterID,
+			IPAddress: exporterIpAddress,
+			FlowType:  string(flow.FlowType),
+		}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** Any chance that a given Netflow exporter is sending multiple flow types?
‏

Reply:
I think it's not common, but you are right, this is I think possible. Will push a fix asap.

Reply:
@FlorianVeaux Fixed here:
- https://github.com/DataDog/datadog-agent/pull/16360/commits/160211c88d6bf282572b6b011da73e8658a33ac8
- tested here https://github.com/DataDog/datadog-agent/pull/16360/commits/5c16f8c9f8d862ba8df840345025ffdcb972accb